qiskit/algorithms/minimum_eigensolvers/numpy_minimum_eigensolver.py

```diff
@@ -0,0 +1,105 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The NumPy minimum eigensolver algorithm and result."""
+
+from __future__ import annotations
+
+from typing import Callable
+import logging
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+# TODO this path will need updating
+from ..eigen_solvers.numpy_eigen_solver import NumPyEigensolver
+from .minimum_eigensolver import MinimumEigensolver, MinimumEigensolverResult
+from ..list_or_dict import ListOrDict
+
+logger = logging.getLogger(__name__)
+
+
+class NumPyMinimumEigensolver(MinimumEigensolver):
+    """
+    The NumPy minimum eigensolver algorithm.
+    """
+
+    def __init__(
+        self,
+        filter_criterion: Callable[
+            [list | np.ndarray, float, ListOrDict[float] | None], bool
+        ] = None,
+    ) -> None:
+        """
+        Args:
+            filter_criterion: callable that allows to filter eigenvalues/eigenstates. The minimum
+                eigensolver is only searching over feasible states and returns an eigenstate that
+                has the smallest eigenvalue among feasible states. The callable has the signature
+                ``filter(eigenstate, eigenvalue, aux_values)`` and must return a boolean to indicate
+                whether to consider this value or not. If there is no feasible element, the result
+                can even be empty.
+        """
+        self._eigensolver = NumPyEigensolver(filter_criterion=filter_criterion)
+
+    @property
+    def filter_criterion(
+        self,
+    ) -> Callable[[list | np.ndarray, float, ListOrDict[float] | None], bool] | None:
+        """Returns the criterion for filtering eigenstates/eigenvalues."""
+        return self._eigensolver.filter_criterion
+
+    @filter_criterion.setter
+    def filter_criterion(
+        self,
+        filter_criterion: Callable[[list | np.ndarray, float, ListOrDict[float] | None], bool],
```

Comment:
The callable `Callable[[list | np.ndarray, float, ListOrDict[float] | None], bool]` appears several times in this file. It would be comprehensive to introduce a type alias.